### PR TITLE
#1991: improve windows helper mock

### LINKED ISSUE #1991: Improve WindowsHelperMock
### Task summary

From PR #1876 we have a left-over from the review that I now decided to move to a new story to unblock the PR from merge:

Improve `WindowsHelperMock` such that a test can simulate to "set registry values" as a preparation for the test so that reading from the windows registry via `WindowsHelper` will give that expected results back that have been prepared.

To do so the `WindowsHelperMock` should simply contain a `Map<String, String>` to mock the windows registry.
Then a test can prepare like this:
```
String appName= "TestApp";
WindowsHelperMock windowsHelperMock = (WindowsHelperMock) context.getWindowsHelper();
windowsHelperMock.setDisplayVersionFromRegistry(appName, "1.1.1");
windowsHelperMock.setDisplayIconFromRegistry(appName, "C:\\Program Files\\TestApp\\testapp.exe,0");
windowsHelperMock.setUninstallStringFromRegistry(appName, "\"C:\\Program Files\\TestApp\\uninstall.exe\"");
windowsHelperMock.setInstallLocationFromRegistry(appName, "C:\\Program Files\\TestApp");
```

I would even go one step further, an improve the design:
* create a record (or class) `WindowsAppInstallation` with the properties `version`, `icon`, `uninstall`, and `install`
* refactor API of `WindowsHelper` so that instead of 4 `get*FromRegistry` methods we just have one method:
```
WindowsAppInstallation getAppInstallationFromRegistry(String appName);
```
* then also the mocking gets much easier since we only need one setter instead of 4 and can use a `Map<String, WindowsAppInstallation>`.

Then we can also get rid of `WindowsHelperImplTestable` again and get everything into `WindowsHelperMock`.

### Additional context

See PR #1876

### PR DESCRIPTION
**Pull request by vivu001**
### Implemented changes:

* Refactor Windows registry API - add WindowsAppApplication and unified lookup
* Improve WindowsHelperMock to support programmable test registry
* Add WindowsHelperImpl tests for registry parsing and edge cases 

---

### Checklist for this PR

Make sure everything is checked before merging this PR. For further info please also see
our [DoD](https://github.com/devonfw/IDEasy/blob/main/documentation/contributing/DoD.adoc).

- [x] When running `mvn clean test` locally all tests pass and build is successful
- [x] PR title is of the form `#«issue-id»: «brief summary»` (e.g. `#921: fixed setup.bat`). If no issue ID exists, title only.
- [x] PR top-level comment summarizes what has been done and contains link to addressed issue(s)
- [x] PR and issue(s) have suitable labels
- [x] Issue is set to `In Progress` and assigned to you *or* there is no issue (might happen for very small PRs)
- [x] You followed all [coding conventions](https://github.com/devonfw/IDEasy/blob/main/documentation/coding-conventions.adoc)
- [x] You have added the issue implemented by your PR in [CHANGELOG.adoc](https://github.com/devonfw/IDEasy/blob/main/CHANGELOG.adoc) unless issue is labeled
  with `internal`

### CODE REVIEW

File: cli/src/test/java/com/devonfw/tools/ide/os/WindowsHelperImplTest.java [hunk 1]
```diff
@@ -52,47 +51,120 @@ void testWindowsHelperParseEmptyRegStringReturnsNull() {
     assertThat(regString).isNull();
   }
 
+  @Test
+  @DisplayName("Test if WindowsAppInstallation can be properly retrieved from registry mock")
```
Comment (hohwille) [inline cli/src/test/java/com/devonfw/tools/ide/os/WindowsHelperImplTest.java]: Using `@DisplayName` replaces the method name in the JUnit output with the given text.
This has advantages and disadvantages.
IMHO we should discuss with the team if we really want to introduce it and use it consistently or not.
Adding a mixture of few methods using it and others not is not homogenous.

File: cli/src/main/java/com/devonfw/tools/ide/os/WindowsHelperImpl.java [hunk 2]
```diff
@@ -74,26 +74,51 @@ public String getRegistryValue(String path, String key) {
...
+    return new WindowsAppInstallation(version, icon, uninstallString, installLocation);
+  }
+
+  @Deprecated(forRemoval = true)
   @Override
   public String getDisplayVersionFromRegistry(String appName) {
     return getRegistryValueBySearch(appName, "DisplayVersion");
   }
 
+  @Deprecated(forRemoval = true)
```
Comment (hohwille) [inline cli/src/main/java/com/devonfw/tools/ide/os/WindowsHelperImpl.java]: Deprecation should be declared in the API (`WindowsHelper`) and not in the implementation to make it visible for the caller.
Do we still have usages or could we just remove these methods directly from the API so we do not need another cleanup PR afterwards?

**Review by hohwille — APPROVED** [review]
@vivu001 thanks for your PR. Nice job. This looks really good now. 👍 IMHO this will improve our tests on the long run.
I only have some minor comments but functionally all is already perfect. Please have a look.

**Comment by vivu001** [comment]
**Successfully retested with version 2026.06.001-06_17_03-SNAPSHOT.**

**Executed:**
mvn -pl cli -am clean test

**Result:**
SUCCESS on Windows (Git Bash)

Behavior is consistent and works as expected.

QA approved.


```
[INFO] Results:
[INFO]
[WARNING] Tests run: 724, Failures: 0, Errors: 0, Skipped: 17
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] ide dev-SNAPSHOT ................................... SUCCESS [  0.515 s]
[INFO] ide-cli 2026.06.001-SNAPSHOT ....................... SUCCESS [03:39 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:40 min
```
